File: backend/xingyun-settle/src/main/java/com/lframework/xingyun/settle/bo/sheet/customer/CustomerSaleSettleInfoBo.java
package com.lframework.xingyun.settle.bo.sheet.customer;

import io.swagger.annotations.ApiModelProperty;
import java.io.Serializable;
import java.math.BigDecimal;
import lombok.Data;

/**
 * 客户销售业务单据结算工作台信息。
 */
@Data
public class CustomerSaleSettleInfoBo implements Serializable {

  private static final long serialVersionUID = 1L;

  @ApiModelProperty("单据ID")
  private String id;

  @ApiModelProperty("业务类型")
  private Integer bizType;

  @ApiModelProperty("单号")
  private String code;

  @ApiModelProperty("客户ID")
  private String customerId;

  @ApiModelProperty("客户名称")
  private String customerName;

  @ApiModelProperty("单据金额")
  private BigDecimal totalAmount;

  @ApiModelProperty("已收金额")
  private BigDecimal receivedAmount;

  @ApiModelProperty("已结算金额")
  private BigDecimal settleAmount;

  @ApiModelProperty("未结算金额")
  private BigDecimal unSettleAmount;

  @ApiModelProperty("结算状态")
  private Integer settleStatus;
}

File: frontend/src/api/customer-settle/sheet/model/customerSaleSettleInfoBo.ts
/** 客户销售结算工作台行数据。 */
export interface CustomerSaleSettleInfoBo {
  /** 单据 ID。 */
  id: string;

  /** 业务类型：1-销售出库单，2-销售退货单。 */
  bizType: number;

  /** 单据号。 */
  code: string;

  /** 客户 ID。 */
  customerId: string;

  /** 客户名称。 */
  customerName: string;

  /** 应收金额。 */
  totalAmount: number;

  /** 已收金额。 */
  receivedAmount: number;

  /** 已对账金额。 */
  checkAmount?: number;

  /** 对账时间。 */
  checkTime?: string;

  /** 对账备注。 */
  checkDescription?: string;

  /** 已结算金额。 */
  settleAmount: number;

  /** 未结算金额。 */
  unSettleAmount: number;

  /** 结算状态。 */
  settleStatus: number;
}
